avocado/core/runner.py

```diff
@@ -468,6 +469,12 @@ def _iter_variants(template, mux):
         :raises ValueError: When variant and template declare params.
         """
         for variant, params in mux.itertests():
+            variant_name = ''
+            for x in params[0]:
+                path = x.path[len('/run'):]
+                variant_name += path.replace('/', '-')
+            # trim first '-'
+            variant_name = variant_name[1:]
```

Comment:
What do I get from all those comments?  That we're maturing a very interesting feature.  Anyway, as @dmonakhov has done before, even if this exact version doesn't get merged, it may lead to an expedited discussion and implementation of a even better version of the proposal.

@ldoktor if you think that it makes sense, let's increase the prioprity of the approriate cards, send small RFCs to the mailing list and follow on with targeted PRs.